R/uvhydrograph-fnc.R

```diff
@@ -1,133 +1,131 @@
 
 #'@export
 uvhydrographPlot <- function(data){
-  uv_pts <- getUvHydro(data, "primarySeries" )
   layout_uvhydro()
   
+  #legend vector, needs to be dynamically built up with plots
+  primary_legend <- getNewLegendFrame()
+  addToPrimaryLegend <- function(newText, newSymbol, newColor, newLine) { 
+    primary_legend <<- rbind(primary_legend, data.frame(text = newText, symbol = newSymbol, color = newColor, line = newLine, stringsAsFactors = FALSE))
```

Comment:
Looking into this a bit more, this is actually not as bad as I first thought.  This is an interesting pattern that might be useful elsewhere.